### DISCUSSION TITLE
Multiple buffers with io_uring_prep_recvmsg_multishot

### DISCUSSION THREAD

**Discussion by ollie-etl** [axboe/liburing#806] [Q&A]
Does `io_uring_prep_recvmsg_multishot` support receiving into multiple buffers, i.e, when `msghdr.msg_iovlen > 1`. I can't see any tests or examples based on this, but also don't see any prose prohibiting this.

**Comment by isilence** [comment]
Multishot receive requests work only with provided buffers, see `IOSQE_BUFFER_SELECT`, which taking out provided buffers from the group specified in `sqe->buf_group` and keep filling them in with received data. And so they don't take any real iovec, but rather some additional information disguised as such. IIRC, for non-multishot `msg_iovlen` should be strictly 1, and only `iovec[0].iov_len` is used to figure out the asked receive size. Don't remember for multishot, but it probably needs `msg_iovlen == 0`.

In short, it doesn't take a normal iovec, and it already can fill multiple buffers that it selects from the specified provided buffer pool.
